libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -493,8 +493,16 @@ bool AP_Arming::arm_checks(uint8_t method)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_LOGGING)) {
-        if (!DataFlash_Class::instance()->logging_started()) {
+        // allow logging to start by indicating (falsely) that the vehicle is armed:
+        DataFlash_Class *df = DataFlash_Class::instance();
+        df->set_vehicle_armed(true);
```

Comment:
I'm not its greatest fan either!

This is currently supposed to be the *last* arm check - at the moment I don't think we have another arm check which changes system state.  So if the logging succeeds, we will arm.

Logging is a particular problem in that if I double-down on lying to dataflash and tell it we're now disarmed - we may rotate the log file (`LOG_FILE_DSRMROT`) :-)  So I think it may always need to be the last arming check.

I guess we could add post-arm checks, but that's a whole new can of worms.

I did notice, however, that we don't seem to make this check for Copter.   Adding a patch for that.